BZ-18892: feat: add an option to include icon within select component

adarshjuspay opened the pull request:
- add an option to include icon within the select component
 - implement CSS styles to ensure proper alignment and display of icon

murdore commented on src/lib/Select/Select.svelte:
```diff
@@ -139,6 +141,11 @@
       role="button"
       tabindex="0"
     >
+    {#if properties.icon && properties.iconPath}
```
add empty string check or default to null above

adarshjuspay replied:
done

murdore commented on src/lib/Select/Select.svelte:
```diff
@@ -139,6 +141,11 @@
     >
+    {#if properties.icon && properties.iconPath}
+      <div class="icon-container">
+        <img src={properties.iconPath} class="icon" alt="" />
```
expose alt via input

adarshjuspay replied:
done

murdore commented on src/lib/Select/Select.svelte:
```diff
@@ -216,8 +223,8 @@
   .arrow {
-    height: 16px;
-    width: 16px;
+    height:var(--arrow-height, 16px);
```
rename to dropdown-icon-height and width

adarshjuspay replied:
done

murdore commented on src/lib/Select/Select.svelte:
```diff
@@ -13,7 +13,9 @@
-    selectMultipleItems: false
+    selectMultipleItems: false,
+    icon:false,
+    iconPath:''
```
do we need to props?

adarshjuspay replied:
resolved

murdore commented:
@adarshjuspay design is not matching with attached screenshot

adarshjuspay commented:
> @adarshjuspay design is not matching with attached screenshot

dev proof updated

rab1prasad commented:
This pr has 2 commits, can we do this here?

rab1prasad commented on src/lib/Select/Select.svelte:
```diff
@@ -6,14 +6,16 @@
 
   let selectedElementDiv: HTMLDivElement | null = null;
 
+  export let alt='';
```
Can we have this `alt` text in `SelectProperties`?

adarshjuspay replied:
we can take alt directly as component prop

rab1prasad commented on src/lib/Select/properties.ts:
```diff
@@ -6,4 +6,5 @@ export type SelectProperties = {
   selectedItemLabel: string | string[] | null;
   showSelectedItemInDropdown: boolean;
   selectMultipleItems: boolean;
+  iconPath:string;
```
```suggestion
  iconPath:string | null;
```

rab1prasad commented on src/lib/Select/Select.svelte:
```diff
@@ -139,6 +141,11 @@
       role="button"
       tabindex="0"
     >
+    {#if properties.iconPath.length!==0}
```
```suggestion
    {#if properties.iconPath !== null && properties.iconPath !== ""}
```

murdore commented on src/lib/Select/Select.svelte:
```diff
@@ -141,9 +141,9 @@
       tabindex="0"
     >
-    {#if properties.icon && properties.iconPath}
+    {#if properties.iconPath.length!==0}
```
add null check

adarshjuspay committed:
BZ-18892: feat: add an option to include icon within select component

    - add an option to include icon within the select component
    - implement CSS styles to ensure proper alignment and display of icon
